p2p/transport/quic/transport.go

```diff
@@ -38,17 +39,116 @@ var quicDialContext = quic.DialContext // so we can mock it in tests
 
 var HolePunchTimeout = 5 * time.Second
 
+// The higher the value, the more accurate the result, but more resources will be used
+const bufferLength = 3
+
+type bufferEntry struct {
+	failedRetries, retries, total uint
+}
+
+type accept struct {
+	buffer           [bufferLength]bufferEntry
+	mtx              sync.Mutex
+	duration         time.Duration
+	fraction         float64
+	start            sync.Once
+	startTime        time.Time // Time of the first execution
+	lastCycle        int64     // last period id (incremental)
+	retryProbability float64   // probability for the Retry packet in the actual time window
+}
+
+func (a *accept) setOptions(duration time.Duration, fraction float64) *accept {
```

Comment:
Probably nicer to have a constructor instead: `newAccept(duration time.Duration, fraction float64)`.